invenio_base/utils.py

```diff
@@ -2,16 +2,44 @@
 #
 # This file is part of Invenio.
 # Copyright (C) 2017-2020 CERN.
+# Copyright (C) 2025 Graz University of Technology.
 #
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Base utilities."""
 
+import importlib.metadata as m
+from collections.abc import Callable
+from sys import version_info
+from typing import Union
+
 from flask import current_app
 from werkzeug.utils import import_string
 
 
+def entry_points(
+    entry_point_group: str,
+    func: Callable = None,
```

Comment:
minor: In general, I'm not a fan of callbacks since they can put downstream users in a particular kind of calling pattern that becomes difficult to read/parse. Was that required for existing use-cases?

Reply:
no it was not, the idea was that the function does not only works as a wrapper around importlib.metadata.entry_points. i wanted to put little bit more functionality in it. so i can remove it again